### PR TITLE
[TS] LPS-172528 Root Folder configuration in the Document & Media widget is lost when using Category Filter.

### PR DESCRIPTION
**Pull request by tototrinh**
Hi,
Please help review this PR and leave your comments. Thanks.
Ticket: https://issues.liferay.com/browse/LPS-172528.
cc @thienquach87

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java [hunk 1]
```diff
@@ -708,9 +711,17 @@ else if (Objects.equals(getOrderByCol(), "size")) {
 								assetEntry.getClassName(),
 								DLFileEntryConstants.getClassName())) {
 
-							results.add(
+							FileEntry fileEntry =
 								DLAppLocalServiceUtil.getFileEntry(
-									assetEntry.getClassPK()));
+									assetEntry.getClassPK());
+
+							if ((folderId ==
```
Comment (AliciaGarciaGarcia) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java:718]: Hi @tototrinh 

We have a similar problem if we select an asset library as the root folder:

with the same steps as described on https://issues.liferay.com/browse/LPS-172528

add this

14. Add an Asset library and link to the site
15. Add documents to the Asset Library
16. On the D&M widget select the asset library home folder as the root folder
17. Select test's Category1 on Category Filter

**Actual result**
 FileA and FileB are shown. 

**Expected result**
Nothing is shown 


So I suggest this code similar to this in the if:
```

	if (_isAncestorFolder(folderId, fileEntry) ||
		((folderId ==
			DLFolderConstants.
				DEFAULT_PARENT_FOLDER_ID) &&
		 (fileEntry.getRepositoryId() ==
			 getRepositoryId()))) {

		results.add(fileEntry);
	}
```

Thnaks!

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

**Comment by thienquach87** [comment]
@AliciaGarciaGarcia
I added your condition and test with both cases, yours and mine.
It works well.
Thanks

**Comment by thienquach87** [comment]
ci:test:sf

**Comment by thienquach87** [comment]
ci:test:relevant

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by Jeremy-Ch** [comment]
The failures are not related to this pr.

**Comment by Jeremy-Ch** [comment]
ci:test:relevant

**Comment by Jeremy-Ch** [comment]
ci:forward

**Pull request closed without merging**